### PR TITLE
From module4 to module3

### PR DESCRIPTION
**Pull request by AleksGF**
Changes from module4 to module3

### CODE REVIEW

File: src/store/appSlice.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  isInitialized: false,
+  isLoading: false,
+  isDetailsLoading: false,
+};
+
+export const appSlice = createSlice({
+  name: 'app',
+  initialState,
+  reducers: {
+    setInitializationStatus: (state, action: PayloadAction<boolean>) => ({
```
Comment (AnAtoliy-AA) [inline src/store/appSlice.ts:19]: action = true

File: src/components/PersonDetails/PersonDetails.tsx [hunk 2]
```diff
@@ -47,16 +40,21 @@ const PersonDetails: FC = () => {
...
+        {!!data && (
+          <>
+            <h3>Person Details:</h3>
+            <p>{`${data.name} was born in ${data.birth_year || 'n/a'}.`}</p>
+            <p>{`${
+              data.gender === 'male'
+                ? 'His'
+                : data.gender === 'female'
+                ? 'Her'
+                : 'Its'
```
Comment (AnAtoliy-AA) [inline src/components/PersonDetails/PersonDetails.tsx:53]: not good

File: src/helpers/getApiPageNumber.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { FIRST_PAGE, ITEMS_PER_PAGE } from '@src/constants/constants';
+
+export const getApiPageNumber = (
+  pageNumber: number | undefined,
```
Comment (AnAtoliy-AA) [inline src/helpers/getApiPageNumber.ts:4]: `pageNumber?: number,`

File: src/services/api/apiClient.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+const prepareHeaders = (headers: Headers): Headers => {
+  headers.set('Content-Type', 'application/json');
+
+  return headers;
+};
+
+export const apiClient = createApi({
+  reducerPath: 'api',
+  baseQuery: fetchBaseQuery({ baseUrl, prepareHeaders }),
+  keepUnusedDataFor: 0,
```
Comment (AnAtoliy-AA) [inline src/services/api/apiClient.ts:14]: magic number

File: src/services/api/peopleApi.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        );
+
+        if (firstResult.error)
+          return { error: firstResult.error as FetchBaseQueryError };
+
+        const data = firstResult.data as PeopleApiResponse;
+        let restQueriesCount = itemsPerPage / ITEMS_PER_PAGE.DEFAULT - 1;
+        let nextPageURL = data.next;
+
+        while (nextPageURL && restQueriesCount > 0) {
```
Comment (AnAtoliy-AA) [inline src/services/api/peopleApi.ts:44]: `while (nextPageURL && restQueriesCount) {`

Comment (severgun) [reply src/services/api/peopleApi.ts:44]: Maybe strict comparison is better for numbers. Negative valuses can be truthy.

**Pull request closed without merging**